src/Rule/Nested.php

```diff
@@ -81,30 +79,35 @@ protected function validateValue($value, ValidationContext $context = null): Res
                 'Value should be an array or an object. %s given.',
                 gettype($value)
             ));
+
             return $result;
         }
+
         $value = (array) $value;
 
         foreach ($this->rules as $valuePath => $rules) {
-            $rulesSet = is_array($rules) ? $rules : [$rules];
             if ($this->errorWhenPropertyPathIsNotFound && !ArrayHelper::pathExists($value, $valuePath)) {
-                $result->addError(
-                    $this->formatMessage(
-                        $this->propertyPathIsNotFoundMessage,
-                        [
-                            'path' => $valuePath,
-                        ]
-                    )
-                );
+                $message = $this->formatMessage($this->propertyPathIsNotFoundMessage, ['path' => $valuePath]);
+                $result->addError($message);
+
                 continue;
             }
+
+            $ruleSet = is_array($rules) ? $rules : [$rules];
+            $aggregatedRule = new Rules($ruleSet);
             $validatedValue = ArrayHelper::getValueByPath($value, $valuePath);
-            $aggregateRule = new Rules($rulesSet);
-            $itemResult = $aggregateRule->validate($validatedValue);
-            if ($itemResult->isValid() === false) {
-                foreach ($itemResult->getErrors() as $error) {
-                    $result->addError($error);
+            $itemResult = $aggregatedRule->validate($validatedValue);
+            if ($itemResult->isValid()) {
+                continue;
+            }
+
+            foreach ($itemResult->getErrorObjects() as $error) {
+                $errorValuePath = is_array($valuePath) ? $valuePath : explode('.', $valuePath);
```

Comment:
How can it be? Could you provide an example?

Reply:
> How can it be? Could you provide an example?

When attribute name is number. Try add rules to '0' => ...

Reply:
Indeed. Not so evident in case of using `Nested`. Maybe add this to the docs as well? :thinking: By the way I caught something like this during debug at some point and then it was gone. :astonished: